### PR TITLE
planner_cspace: add watchdog to costmap update

### LINKED ISSUE #228: Check incoming data rate in costmap_3d and planner_3d
stop if data rate is unexpected and publish error status

### PR DESCRIPTION
**Pull request by at-wat**
fix #228 

need https://github.com/at-wat/neonavigation_msgs/pull/7

### CODE REVIEW

**Comment by at-wat** [comment]
## [#658] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/costmap_cspace/gtest-test_costmap_3d.xml: 6 tests
build/test_results/costmap_cspace/roslint-costmap_cspace.xml: 1 tests
build/test_results/joystick_interrupt/roslint-joystick_interrupt.xml: 1 tests
build/test_results/joystick_interrupt/rostest-test_test_joystick_interrupt_rostest.xml: 1 tests
build/test_results/joystick_interrupt/rosunit-test_joystick_interrupt.xml: 3 tests
build/test_results/map_organizer/roslint-map_organizer.xml: 1 tests
build/test_results/neonavigation_common/roslint-neonavigation_common.xml: 1 tests
build/test_results/obj_to_pointcloud/roslint-obj_to_pointcloud.xml: 1 tests
build/test_results/planner_cspace/gtest-test_blockmem_gridmap.xml: 3 tests
build/test_results/planner_cspace/gtest-test_cyclic_vec.xml: 7 tests
build/test_results/planner_cspace/gtest-test_grid_metric_converter.xml: 2 tests
build/test_results/planner_cspace/gtest-test_motion_cache.xml: 1 tests
build/test_results/planner_cspace/roslint-planner_cspace.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_abort_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_costmap_watchdog_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_compat_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_preempt_rostest.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_abort.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_costmap_watchdog.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_navigate.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_preempt.xml: 1 tests
build/test_results/safety_limiter/roslint-safety_limiter.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_compat_rostest.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_rostest.xml: 1 tests
build/test_results/safety_limiter/rosunit-test_safety_limiter.xml: 7 tests
build/test_results/track_odometry/gtest-test_tf_projection.xml: 1 tests
build/test_results/track_odometry/roslint-track_odometry.xml: 1 tests
build/test_results/track_odometry/rostest-test_test_track_odometry_rostest.xml: 1 tests
build/test_results/track_odometry/rosunit-test_track_odometry.xml: 1 tests
build/test_results/trajectory_tracker/gtest-test_trajectory_tracker_filter.xml: 3 tests
build/test_results/trajectory_tracker/roslint-trajectory_tracker.xml: 1 tests
build/test_results/trajectory_tracker/rostest-test_test_trajectory_recorder_rostest.xml: 1 tests
build/test_results/trajectory_tracker/rosunit-test_trajectory_recorder.xml: 1 tests
Summary: 58 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#659] PASSED on kinetic

<details><summary>All tests passed</summary>


```
build/test_results/costmap_cspace/gtest-test_costmap_3d.xml: 6 tests
build/test_results/costmap_cspace/roslint-costmap_cspace.xml: 1 tests
build/test_results/joystick_interrupt/roslint-joystick_interrupt.xml: 1 tests
build/test_results/joystick_interrupt/rostest-test_test_joystick_interrupt_rostest.xml: 1 tests
build/test_results/joystick_interrupt/rosunit-test_joystick_interrupt.xml: 3 tests
build/test_results/map_organizer/roslint-map_organizer.xml: 1 tests
build/test_results/neonavigation_common/roslint-neonavigation_common.xml: 1 tests
build/test_results/obj_to_pointcloud/roslint-obj_to_pointcloud.xml: 1 tests
build/test_results/planner_cspace/gtest-test_blockmem_gridmap.xml: 3 tests
build/test_results/planner_cspace/gtest-test_cyclic_vec.xml: 7 tests
build/test_results/planner_cspace/gtest-test_grid_metric_converter.xml: 2 tests
build/test_results/planner_cspace/gtest-test_motion_cache.xml: 1 tests
build/test_results/planner_cspace/roslint-planner_cspace.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_abort_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_costmap_watchdog_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_compat_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_preempt_rostest.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_abort.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_costmap_watchdog.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_navigate.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_preempt.xml: 1 tests
build/test_results/safety_limiter/roslint-safety_limiter.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_compat_rostest.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_rostest.xml: 1 tests
build/test_results/safety_limiter/rosunit-test_safety_limiter.xml: 7 tests
build/test_results/track_odometry/gtest-test_tf_projection.xml: 1 tests
build/test_results/track_odometry/roslint-track_odometry.xml: 1 tests
build/test_results/track_odometry/rostest-test_test_track_odometry_rostest.xml: 1 tests
build/test_results/track_odometry/rosunit-test_track_odometry.xml: 1 tests
build/test_results/trajectory_tracker/gtest-test_trajectory_tracker_filter.xml: 3 tests
build/test_results/trajectory_tracker/roslint-trajectory_tracker.xml: 1 tests
build/test_results/trajectory_tracker/rostest-test_test_trajectory_recorder_rostest.xml: 1 tests
build/test_results/trajectory_tracker/rosunit-test_trajectory_recorder.xml: 1 tests
Summary: 58 tests, 0 errors, 0 failures, 0 skipped
```
</details>

File: Dockerfile.indigo [hunk 1]
```diff
@@ -28,7 +28,7 @@ COPY ./trajectory_tracker/package.xml /catkin_ws/src/neonavigation/trajectory_tr
 COPY ./obj_to_pointcloud/package.xml /catkin_ws/src/neonavigation/obj_to_pointcloud/
 COPY ./track_odometry/package.xml /catkin_ws/src/neonavigation/track_odometry/
 
-RUN cd /catkin_ws/src && git clone --depth=1 https://github.com/at-wat/neonavigation_msgs.git
+RUN cd /catkin_ws/src && git clone --depth=1 -b add-more-planner-error-status https://github.com/at-wat/neonavigation_msgs.git
```
Comment (at-wat) [inline Dockerfile.indigo]: revert before merge

File: planner_cspace/src/planner_3d.cpp [hunk 2]
```diff
@@ -1796,17 +1830,25 @@ class Planner3dNode
...
-    else if (status_.error == planner_cspace_msgs::PlannerStatus::PATH_NOT_FOUND)
+    switch (status_.error)
     {
-      stat.summary(diagnostic_msgs::DiagnosticStatus::ERROR, "Path not found.");
-    }
-    else
-    {
-      stat.summary(diagnostic_msgs::DiagnosticStatus::OK, "Going well.");
+      case planner_cspace_msgs::PlannerStatus::GOING_WELL:
+        break;
```
Comment (ykoga-kyutech) [inline planner_cspace/src/planner_3d.cpp]: @at-wat 
No message shown in /diagnostics if PlannerStatus::GOING_WELL?

- going well
```
$ rostopic echo /diagnostics

header: 
  seq: 107
  stamp: 
    secs: 1543971395
    nsecs: 467201380
  frame_id: ''
status: 
  - 
    level: 2
    name: "planner_3d: Path Planner Status"
    message: "No message was set"
    hardware_id: "none"
    values: 
      - 
        key: "status"
        value: "1"
      - 
        key: "error"
        value: "0"
---

```

- cf. path not found
```
---
header: 
  seq: 278
  stamp: 
    secs: 1543971505
    nsecs: 666996955
  frame_id: ''
status: 
  - 
    level: 2
    name: "planner_3d: Path Planner Status"
    message: "Path not found."
    hardware_id: "none"
    values: 
      - 
        key: "status"
        value: "1"
      - 
        key: "error"
        value: "2"
---
```

**Comment by at-wat** [comment]
## [#660] FAILED on indigo

docker build failed

**Comment by at-wat** [comment]
## [#660] FAILED on kinetic

```catkin_make``` failed

File: planner_cspace/src/planner_3d.cpp (hunk 2 above)
Comment (at-wat) [reply planner_cspace/src/planner_3d.cpp]: I have added a check for DATA_MISSING and GOING_WELL diagnostics status in the test.
Test for the other cases of diagnostics outputs might better be added.

**Comment by at-wat** [comment]
## [#662] FAILED on indigo

<details><summary>Test failed</summary>


```
build/test_results/costmap_cspace/gtest-test_costmap_3d.xml: 6 tests
build/test_results/costmap_cspace/roslint-costmap_cspace.xml: 1 tests
build/test_results/joystick_interrupt/roslint-joystick_interrupt.xml: 1 tests
build/test_results/joystick_interrupt/rostest-test_test_joystick_interrupt_rostest.xml: 1 tests
build/test_results/joystick_interrupt/rosunit-test_joystick_interrupt.xml: 3 tests
build/test_results/map_organizer/roslint-map_organizer.xml: 1 tests
build/test_results/neonavigation_common/roslint-neonavigation_common.xml: 1 tests
build/test_results/obj_to_pointcloud/roslint-obj_to_pointcloud.xml: 1 tests
build/test_results/planner_cspace/gtest-test_blockmem_gridmap.xml: 3 tests
build/test_results/planner_cspace/gtest-test_cyclic_vec.xml: 7 tests
build/test_results/planner_cspace/gtest-test_grid_metric_converter.xml: 2 tests
build/test_results/planner_cspace/gtest-test_motion_cache.xml: 1 tests
build/test_results/planner_cspace/roslint-planner_cspace.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_abort_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_costmap_watchdog_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_compat_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_preempt_rostest.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_abort.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_costmap_watchdog.xml: 1 tests, 0 errors, 1 failures, 0 skipped
build/test_results/planner_cspace/rosunit-test_navigate.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_preempt.xml: 1 tests
build/test_results/safety_limiter/roslint-safety_limiter.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_compat_rostest.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_rostest.xml: 1 tests
build/test_results/safety_limiter/rosunit-test_safety_limiter.xml: 7 tests
build/test_results/track_odometry/gtest-test_tf_projection.xml: 1 tests
build/test_results/track_odometry/roslint-track_odometry.xml: 1 tests
build/test_results/track_odometry/rostest-test_test_track_odometry_rostest.xml: 1 tests
build/test_results/track_odometry/rosunit-test_track_odometry.xml: 1 tests
build/test_results/trajectory_tracker/gtest-test_trajectory_tracker_filter.xml: 3 tests
build/test_results/trajectory_tracker/roslint-trajectory_tracker.xml: 1 tests
build/test_results/trajectory_tracker/rostest-test_test_trajectory_recorder_rostest.xml: 1 tests
build/test_results/trajectory_tracker/rosunit-test_trajectory_recorder.xml: 1 tests
Summary: 58 tests, 0 errors, 1 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#663] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/costmap_cspace/gtest-test_costmap_3d.xml: 6 tests
build/test_results/costmap_cspace/roslint-costmap_cspace.xml: 1 tests
build/test_results/joystick_interrupt/roslint-joystick_interrupt.xml: 1 tests
build/test_results/joystick_interrupt/rostest-test_test_joystick_interrupt_rostest.xml: 1 tests
build/test_results/joystick_interrupt/rosunit-test_joystick_interrupt.xml: 3 tests
build/test_results/map_organizer/roslint-map_organizer.xml: 1 tests
build/test_results/neonavigation_common/roslint-neonavigation_common.xml: 1 tests
build/test_results/obj_to_pointcloud/roslint-obj_to_pointcloud.xml: 1 tests
build/test_results/planner_cspace/gtest-test_blockmem_gridmap.xml: 3 tests
build/test_results/planner_cspace/gtest-test_cyclic_vec.xml: 7 tests
build/test_results/planner_cspace/gtest-test_grid_metric_converter.xml: 2 tests
build/test_results/planner_cspace/gtest-test_motion_cache.xml: 1 tests
build/test_results/planner_cspace/roslint-planner_cspace.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_abort_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_costmap_watchdog_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_compat_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_preempt_rostest.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_abort.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_costmap_watchdog.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_navigate.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_preempt.xml: 1 tests
build/test_results/safety_limiter/roslint-safety_limiter.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_compat_rostest.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_rostest.xml: 1 tests
build/test_results/safety_limiter/rosunit-test_safety_limiter.xml: 7 tests
build/test_results/track_odometry/gtest-test_tf_projection.xml: 1 tests
build/test_results/track_odometry/roslint-track_odometry.xml: 1 tests
build/test_results/track_odometry/rostest-test_test_track_odometry_rostest.xml: 1 tests
build/test_results/track_odometry/rosunit-test_track_odometry.xml: 1 tests
build/test_results/trajectory_tracker/gtest-test_trajectory_tracker_filter.xml: 3 tests
build/test_results/trajectory_tracker/roslint-trajectory_tracker.xml: 1 tests
build/test_results/trajectory_tracker/rostest-test_test_trajectory_recorder_rostest.xml: 1 tests
build/test_results/trajectory_tracker/rosunit-test_trajectory_recorder.xml: 1 tests
Summary: 58 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#664] FAILED on indigo

```catkin_make``` failed

**Comment by at-wat** [comment]
CI failing due to #236

**Comment by at-wat** [comment]
## [#667] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/costmap_cspace/gtest-test_costmap_3d.xml: 6 tests
build/test_results/costmap_cspace/roslint-costmap_cspace.xml: 1 tests
build/test_results/joystick_interrupt/roslint-joystick_interrupt.xml: 1 tests
build/test_results/joystick_interrupt/rostest-test_test_joystick_interrupt_rostest.xml: 1 tests
build/test_results/joystick_interrupt/rosunit-test_joystick_interrupt.xml: 3 tests
build/test_results/map_organizer/roslint-map_organizer.xml: 1 tests
build/test_results/neonavigation_common/roslint-neonavigation_common.xml: 1 tests
build/test_results/obj_to_pointcloud/roslint-obj_to_pointcloud.xml: 1 tests
build/test_results/planner_cspace/gtest-test_blockmem_gridmap.xml: 3 tests
build/test_results/planner_cspace/gtest-test_cyclic_vec.xml: 7 tests
build/test_results/planner_cspace/gtest-test_grid_metric_converter.xml: 2 tests
build/test_results/planner_cspace/gtest-test_motion_cache.xml: 1 tests
build/test_results/planner_cspace/roslint-planner_cspace.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_abort_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_costmap_watchdog_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_compat_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_navigation_rostest.xml: 1 tests
build/test_results/planner_cspace/rostest-test_test_preempt_rostest.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_abort.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_costmap_watchdog.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_navigate.xml: 1 tests
build/test_results/planner_cspace/rosunit-test_preempt.xml: 1 tests
build/test_results/safety_limiter/roslint-safety_limiter.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_compat_rostest.xml: 1 tests
build/test_results/safety_limiter/rostest-test_test_safety_limiter_rostest.xml: 1 tests
build/test_results/safety_limiter/rosunit-test_safety_limiter.xml: 7 tests
build/test_results/track_odometry/gtest-test_tf_projection.xml: 1 tests
build/test_results/track_odometry/roslint-track_odometry.xml: 1 tests
build/test_results/track_odometry/rostest-test_test_track_odometry_rostest.xml: 1 tests
build/test_results/track_odometry/rosunit-test_track_odometry.xml: 1 tests
build/test_results/trajectory_tracker/gtest-test_trajectory_tracker_filter.xml: 3 tests
build/test_results/trajectory_tracker/roslint-trajectory_tracker.xml: 1 tests
build/test_results/trajectory_tracker/rostest-test_test_trajectory_recorder_rostest.xml: 1 tests
build/test_results/trajectory_tracker/rosunit-test_trajectory_recorder.xml: 1 tests
Summary: 58 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Pull request merged**